### PR TITLE
Meg branch local

### PR DESCRIPTION
**Pull request by mgtucks**
Updated Has Clue

### CODE REVIEW

**Comment by mgtucks** [comment]
**whoHasClue not hasClue

File: src/main/java/clueless/Game.java [hunk 1]
```diff
@@ -383,15 +384,38 @@ public void loseGame(Player loser) {
 	 * @return
 	 */
 	public Player whoHasClue(List<Card> suggestion) {
-		Player pl;
+		Player pl = null;
```
Comment (zbialik) [inline src/main/java/clueless/Game.java:387]: rather that initializing and assigning to null and returning `pl` at the end, just initialize it here and if you make it to the end of method return null

File: src/main/java/clueless/Game.java [hunk 2]
```diff
@@ -383,15 +384,38 @@ public void loseGame(Player loser) {
...
+						playerHasClue = true;
+
+					} else 
+					{
+						nextPlayerindex++;
+						count++;
+					}
+			}
+		}
+			return pl;
```
Comment (zbialik) [inline src/main/java/clueless/Game.java:417]:
```suggestion
			return null;
```

File: src/main/java/clueless/Game.java [hunk 3]
```diff
@@ -383,15 +384,38 @@ public void loseGame(Player loser) {
...
-		//		}
-
-		return null;
+		
+			if (nextPlayerindex >= CHARACTER_TURN_ORDER.length) 
+				{ // repoint index if currIndex is last character
+					nextPlayerindex = 0;
+				}
+			
+			while ((!playerHasClue) && i < this.characterMap.size())
```
Comment (zbialik) [inline src/main/java/clueless/Game.java]: you reference i here but never increment `i`, did you mean to use `count` here? If so, you should initialize count outside this while loop

File: src/main/java/clueless/Game.java [hunk 4]
```diff
@@ -383,15 +384,38 @@ public void loseGame(Player loser) {
...
+			{
+				
+				int count = 0;
+				while (count < CHARACTER_TURN_ORDER.length && nextPlayerindex != currIndex) {
+
+					if (this.isPlayer(CHARACTER_TURN_ORDER[nextPlayerindex]) 
+							&& this.getPlayer(CHARACTER_TURN_ORDER[nextPlayerindex]).hasClue(suggestion)) {
+						pl = this.getPlayer(CHARACTER_TURN_ORDER[nextPlayerindex]);
```
Comment (zbialik) [inline src/main/java/clueless/Game.java:407]:
```suggestion
						return this.getPlayer(CHARACTER_TURN_ORDER[nextPlayerindex]);
```

you can just return the player that has a clue here

**Review by zbialik — CHANGES_REQUESTED** [review]
looks good -- just need remediate the counter in that while loop comment

**Pull request closed without merging**